src/Microsoft.DotNet.Cli.Utils/CommandResolution/DepsJsonCommandResolver.cs

```diff
@@ -0,0 +1,261 @@
+
+        public DepsJsonCommandResolver(string nugetPackageRoot)
+        {
+            _muxer = new Muxer();
+            _nugetPackageRoot = nugetPackageRoot;
+        }
+
+        public DepsJsonCommandResolver(Muxer muxer, string nugetPackageRoot)
+        {
+            _muxer = muxer;
+            _nugetPackageRoot = nugetPackageRoot;
+        }
+
+        public CommandSpec Resolve(CommandResolverArguments commandResolverArguments)
+        {
+            if (commandResolverArguments.CommandName == null
+                || commandResolverArguments.DepsJsonFile == null)
+            {
+                return null;
+            }
+            
+            return ResolveFromDepsJsonFile(
+                commandResolverArguments.CommandName, 
+                commandResolverArguments.CommandArguments.OrEmptyIfNull(),
+                commandResolverArguments.DepsJsonFile);
+        }
+
+        private CommandSpec ResolveFromDepsJsonFile(
+            string commandName, 
+            IEnumerable<string> commandArgs, 
+            string depsJsonFile)
+        {
+            var dependencyContext = LoadDependencyContextFromFile(depsJsonFile);
+
+            var commandPath = GetCommandPathFromDependencyContext(commandName, dependencyContext);
+            if (commandPath == null)
+            {
+                return null;
+            }
+
+            return CreateCommandSpecUsingMuxerIfPortable(
+                commandPath, 
+                commandArgs, 
+                depsJsonFile,
+                CommandResolutionStrategy.DepsFile,
+                _nugetPackageRoot,
+                IsPortableApp(commandPath));
+        }
+
+        public DependencyContext LoadDependencyContextFromFile(string depsJsonFile)
+        {
+            DependencyContext dependencyContext = null;
+            DependencyContextJsonReader contextReader = new DependencyContextJsonReader();
+
+            using (var contextStream = File.OpenRead(depsJsonFile))
+            {
+                dependencyContext = contextReader.Read(contextStream);
+            }
+
+            return dependencyContext;
+        }
+
+        public string GetCommandPathFromDependencyContext(string commandName, DependencyContext dependencyContext)
+        {
+            var commandCandidates = new List<CommandCandidate>();
+
+            var assemblyCommandCandidates = GetAssemblyCommandCandidates(commandName, dependencyContext);
+            var nativeCommandCandidates = GetNativeCommandCandidates(commandName, dependencyContext);
+
+            commandCandidates.AddRange(assemblyCommandCandidates);
+            commandCandidates.AddRange(nativeCommandCandidates);
+
+            var command = ChooseCommandCandidate(commandCandidates);
+
+            return command?.GetAbsoluteCommandPath(_nugetPackageRoot);
+        }
+
+        private IEnumerable<CommandCandidate> GetAssemblyCommandCandidates(string commandName, DependencyContext dependencyContext)
```

Comment:
This method and GetNativeCommandCandidates and effectively the same, except the candidates are different (which can be a parameter) and they invoke a different delegate to get the candidates, which can be a parameter as well. How about some refactoring? :smile: